### PR TITLE
Fix Issue #543 EditLink is wrong in 5.7

### PR DESCRIPTION
**Pull request by VikingsFan**
Use default behavior when target is not contained navigation property
When target is contained navigation property, just keep the last non-contained navigation srouce.

### CODE REVIEW

File: OData/test/E2ETest/WebStack.QA.Test.OData/ODataPathHandler/LinkGenerationTests.cs [hunk 1]
```diff
@@ -89,5 +105,19 @@ public void GeneratedLinkShouldMatchRequestRouting()
             content = this.Client.GetStringAsync(this.BaseAddress + "/v2/LinkGeneration_Model2").Result;
             Assert.DoesNotContain(@"/v1/LinkGeneration_Model2", content);
         }
+
+        [Theory]
+        [InlineData("/v1/LinkGeneration_Model1(1)/NonContainedNavigationProperty", "/v1/LinkGeneration_Model2(2)")]
```
Comment (xuzhg) [inline OData/test/E2ETest/WebStack.QA.Test.OData/ODataPathHandler/LinkGenerationTests.cs]: Only one test case, it's better to use [Fact] and put the cast condition into test body.

File: OData/test/E2ETest/WebStack.QA.Test.OData/ODataPathHandler/LinkGenerationTests.cs [hunk 2]
```diff
@@ -89,5 +105,19 @@ public void GeneratedLinkShouldMatchRequestRouting()
...
+        public void GeneratedLinkTestForNavigationProperty(string requestUrl, string expectLinkUrl)
+        {
+            string AcceptJsonFullMetadata = "application/json;odata.metadata=full";
```
Comment (xuzhg) [inline OData/test/E2ETest/WebStack.QA.Test.OData/ODataPathHandler/LinkGenerationTests.cs]: can remove the random variable if it's only used once.

**Comment by xuzhg** [comment]
looks good.

**Comment by VikingsFan** [comment]
Merged 86e222607410a7bb61698b8f2ce69a0b1f24d901

**Pull request closed without merging**